### DISCUSSION TITLE
✨ Daggerverse: modules should display their engine version

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#12397] [Ideas]
### What are you trying to do?

Need to know engine version that a module was built with so can know if a module is compatible with my engine version.

Want to filter for modules by engine version.

### Why is this important to you?

Modules created with an engine version different from the one I'm running might not behave properly.
https://discord.com/channels/707636530424053791/1120503349599543376/1227216892691742761

### How are you currently working around this?

You can look at the `dagger.json` in a module's git repo.

**Comment by jedevc** [comment]
Hm, I agree that this is currently an issue, but I don't think the solution is to encourage users to check the engine version in daggerverse.

IMO, if a `dagger.json` engine version is less than or equal to the engine version, we should *always* be able to run it - for old versions, we can introduce backwards compatibility to provide an old API for old modules. For newer modules... we could just refuse to run them for now? Something like #7031 but for modules?

---

That said, that's a longer term commitment to breaking less and to provide a more stable experience - so maybe temporarily having this info in daggerverse might be a good shout?
